### PR TITLE
N Body energy

### PR DESCRIPTION
**Pull request by davkovacs**
Might not be the most elegant way of doing this, but I think it works. Takes a PIPotential, an Atoms object and a Body order as arguments and returns the contribution to the energy of the correct basis functions.

### CODE REVIEW

**Review by cortner — COMMENTED** [review]
to discuss later

File: Project.toml [hunk 1]
```diff
@@ -25,6 +25,7 @@ ZipFile = "a5390f91-8eb1-5f08-bee0-b1d1ffed6cea"
 
 [compat]
 JuLIP = "0.10.1 - 0"
+JuLIP = "0.11.2"
```
[the remark this answers is not in the document]
Comment (davkovacs) [reply Project.toml:28]: I had 0.11.2 on my machine, but nothing requires it. I should have removed that probably.

File: src/utils/pure.jl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  ind = 1
+  for i in 1:length(coeff_size)
+    for j in 1:coeff_size[i]
+        if Ns_pi[ind] != korr_ord
+            Cs[i][j] = 0
+        end
+        ind += 1
+    end
+  end
+  IP_new = PIPotential(IP.pibasis, Cs)
```
[the remark this answers is not in the document]
Comment (davkovacs) [reply src/utils/pure.jl:41]: Makes sense. I could do that.

**Comment by davkovacs** [comment]
I have cleaned up according to our discussion, ready to merge (except for the JuLIP version part)

File: src/utils/pure.jl [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+include("../extimports.jl")
+include("../aceimports.jl")
+
+using JuLIP
+
+export nB_energy, nB_potential
+
+function nB_energy(IP::PIPotential, atoms::Atoms, B_order::Int)
+  """
```
Comment (cortner) [inline src/utils/pure.jl:16]: doc string goes in front of the function

File: src/utils/pure.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (cortner) [reply src/utils/pure.jl:33]: are you now duplicating code?  I think `nB_energy` should first call `nB_potential` to get the new model and then evaluate the new model.

File: src/utils/pure.jl [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    if ord != korr_ord
+        Cs[i][j] = 0
+    end
+  end
+  IP_new = PIPotential(IP.pibasis, Cs)
+  return energy(IP_new, atoms)
+end
+
+function nB_potential(IP::PIPotential, B_order::Int)
+  """
```
Comment (cortner) [inline src/utils/pure.jl:38]: doc string in front of function

**Review by cortner — CHANGES_REQUESTED** [review]
I few more things. Sorry I'm slow. Too many things to take care of ...

**Comment by davkovacs** [comment]
Thank you for the feedback, and being patient, I am learning how to do all this, as I am doing these changes. I think I have changed everything as requested.

**Comment by cortner** [comment]
sorry I dropped the ball on this.

**Pull request merged**